source/common/quic/quic_client_transport_socket_factory.cc

```diff
@@ -22,19 +34,19 @@ QuicClientTransportSocketConfigFactory::createTransportSocketFactory(
       config, context.messageValidationVisitor());
   auto client_config = std::make_unique<Extensions::TransportSockets::Tls::ClientContextConfigImpl>(
       quic_transport.upstream_tls_context(), context);
-  auto factory =
-      std::make_unique<QuicClientTransportSocketFactory>(std::move(client_config), context);
-  factory->initialize();
-  return factory;
+  return QuicClientTransportSocketFactory::create(std::move(client_config), context);
 }
 
 QuicClientTransportSocketFactory::QuicClientTransportSocketFactory(
     Ssl::ClientContextConfigPtr config,
-    Server::Configuration::TransportSocketFactoryContext& factory_context)
+    Server::Configuration::TransportSocketFactoryContext& factory_context,
+    absl::Status& creation_status)
     : QuicTransportSocketFactoryBase(factory_context.statsScope(), "client"),
-      fallback_factory_(std::make_unique<Extensions::TransportSockets::Tls::ClientSslSocketFactory>(
-          std::move(config), factory_context.sslContextManager(), factory_context.statsScope())),
       tls_slot_(factory_context.serverFactoryContext().threadLocal()) {
+  auto factory_or_error = Extensions::TransportSockets::Tls::ClientSslSocketFactory::create(
```

Comment:
https://abseil.io/tips/181 recommends against the `_or` suffix in the name, and would recommend simply `factory` in this case. Here and elsewhere.

Reply:
I think with heavy use of auto in Envoy I'm not comfortable hiding away the fact this is a statusor so I'd like to argue this adds clarity for us and stick with it.

Reply:
I think this is a good argument for, as the style guide would say, using the explicit type here since it will make the code more clear (and then removing the `_or` suffix as per the abeil guide).
https://google.github.io/styleguide/cppguide.html#Type_deduction